library/include/ck/library/tensor_operation_instance/gpu/grouped_convolution_forward.hpp

```diff
@@ -523,127 +523,200 @@ struct DeviceOperationInstanceFactory<ck::tensor_operation::device::DeviceGroupe
             }
 #endif
         }
-#endif
-
-#ifdef CK_USE_WMMA
-        if constexpr(NumDimSpatial == 2 && is_same_v<InLayout, GNHWC> &&
-                     is_same_v<WeiLayout, GKYXC> && is_same_v<OutLayout, GNHWK>)
+        // layout NGCHW/GKYXC/NGKHW
+        if constexpr(NumDimSpatial == 3 && is_same_v<InLayout, NGCHW> &&
+                     is_same_v<WeiLayout, GKCYX> && is_same_v<OutLayout, NGKHW>)
         {
+#ifdef CK_ENABLE_FP32
+            if constexpr(is_same_v<InDataType, float> && is_same_v<WeiDataType, float> &&
+                         is_same_v<OutDataType, float> && is_same_v<AComputeType, float> &&
+                         is_same_v<BComputeType, float>)
+            {
+                add_device_grouped_conv3d_fwd_xdl_merged_groups_ngchw_gkcyx_ngkhw_f32_instances(
+                    op_ptrs);
+                add_device_grouped_conv3d_fwd_xdl_ngchw_gkcyx_ngkhw_f32_instances(op_ptrs);
+                add_device_grouped_conv3d_fwd_xdl_ngchw_gkcyx_ngkhw_f32_comp_instances(op_ptrs);
+                add_device_grouped_conv3d_fwd_xdl_ngchw_gkcyx_ngkhw_f32_mem_intra_instances(
+                    op_ptrs);
+                add_device_grouped_conv3d_fwd_xdl_ngchw_gkcyx_ngkhw_f32_mem_inter_instances(
+                    op_ptrs);
+            }
+#endif
 #ifdef CK_ENABLE_FP16
             if constexpr(is_same_v<InDataType, half_t> && is_same_v<WeiDataType, half_t> &&
                          is_same_v<OutDataType, half_t> && is_same_v<AComputeType, half_t> &&
                          is_same_v<BComputeType, half_t>)
             {
-                add_device_grouped_conv2d_fwd_wmma_gnhwc_gkyxc_gnhwk_f16_instances(op_ptrs);
-                add_device_grouped_conv2d_fwd_wmma_gnhwc_gkyxc_gnhwk_f16_1x1p0_instances(op_ptrs);
-                add_device_grouped_conv2d_fwd_wmma_gnhwc_gkyxc_gnhwk_f16_1x1s1p0_instances(op_ptrs);
-                add_device_grouped_conv2d_fwd_wmma_gnhwc_gkyxc_gnhwk_f16_oddc_instances(op_ptrs);
+                add_device_grouped_conv3d_fwd_xdl_merged_groups_ngchw_gkcyx_ngkhw_f16_instances(
+                    op_ptrs);
+                add_device_grouped_conv3d_fwd_xdl_ngchw_gkcyx_ngkhw_f16_instances(op_ptrs);
+                add_device_grouped_conv3d_fwd_xdl_ngchw_gkcyx_ngkhw_f16_comp_instances(op_ptrs);
```

Comment:
I dont see cpp files for:
                add_device_grouped_conv3d_fwd_xdl_ngchw_gkcyx_ngkhw_f16_comp_instances(op_ptrs);
                add_device_grouped_conv3d_fwd_xdl_ngchw_gkcyx_ngkhw_f16_comp_2x_instances(op_ptrs);
                add_device_grouped_conv3d_fwd_xdl_ngchw_gkcyx_ngkhw_f16_comp_part2_instances(
                    op_ptrs);
                add_device_grouped_conv3d_fwd_xdl_ngchw_gkcyx_ngkhw_f16_mem_intra_instances(
                    op_ptrs);
                add_device_grouped_conv3d_fwd_xdl_ngchw_gkcyx_ngkhw_f16_mem_inter_instances(
                    op_ptrs);